### DISCUSSION TITLE
feature: add supermaven

### DISCUSSION THREAD

**Discussion by sashalikesplanes** [LazyVim/LazyVim#3471] [Extra Requests]
### Did you check the docs?

- [X] I have read all the LazyVim docs

### Is your feature request related to a problem? Please describe.

A new copilot style plugin that is significantly faster and nicer to use.

https://github.com/supermaven-inc/supermaven-nvim

### Describe the solution you'd like

Add supermaven as an optional plugin similar to Copilot and Codium

### Describe alternatives you've considered

Looked for the list of plugins that are currently available, did not see any issues or PRs mentioning supermaven

**Comment by sashalikesplanes** [comment]
I could have a go at it myself by copying how Copilot plugin is implemented. Let me know if such PR would be welcome

**Comment by dpetka2001** [comment]
I've tried supermaven and in my personal opinion it's still rough around the edges. It seemed definitely faster than Codeium (haven't tried Copilot myself), but it doesn't support multiple suggestions yet (except if they added support since 2-3 weeks ago that I tested it).

In my personal experience as a regular user like you, I'd suggest, for now, that users install it in their personal configuration if they want to.

**Comment by dotfrag** [comment]
I agree with @dpetka2001, I'm using it now and it's really good but sometimes it feels buggy on the completion.

For what it's worth, the setup is very simple:

```lua
return {
  "nvim-cmp",
  dependencies = {
    "supermaven-inc/supermaven-nvim",
    build = ":SupermavenUseFree", -- remove this line if you are using pro
    opts = {},
  },
  ---@param opts cmp.ConfigSchema
  opts = function(_, opts)
    table.insert(opts.sources, 1, {
      name = "supermaven",
      group_index = 1,
      priority = 100,
    })
  end,
}
```

**Reply by occult** [reply]
This worked! thanks a lot

**Comment by iduuck** [comment]
What is the build command especially for? As far as I can see, it's fine to just drop the build command, and then call the following in the opts:

```lua
local api = require("supermaven-nvim.api")

if api.is_running() then
	api.start()
end
```

Or am I wrong?

**Comment by dotfrag** [comment]
The build command is absolutely *not* necessary, but I added it there because, supermaven, when launched for the first time, prompts you to authenticate for the pro version. This prevents that and uses the free version instead. Of course, if you are paying for the pro version, you don't need the build option.

The code snippet you mentioned I believe is not necessary. The code from my comment above already starts it.

**Comment by ghost** [comment]
I have used SuperMaven, and couldn't install it here, so this will be nice

**Comment by rmunn** [comment]
Would a PR to add supermaven as an extra be acceptable? One of my coworkers uses it (in JetBrains Rider) and he likes it, so I want to try it out. I'd love to be able to install it with just an `x` in the extras list. So if a PR would be accepted, I'm willing to do the work to create one.

**Reply by folke** [reply]
Supermaven was added yesterday

**Reply by rmunn** [reply]
I just noticed it in the extras list, came back to say "Um, never mind, I'm blind and didn't see it." But now I feel better: it really *wasn't* there when I posted that comment, because I hadn't run Sync yet.

I believe that's about the speediest "Hey, the feature I asked for has now been added" experience I've ever had on GitHub. :grin: Thank you.

**Reply by RHOOPH** [reply]
I have installed supermaven extra, and have this weird thing happening, 


https://github.com/user-attachments/assets/24c55767-e564-4273-975a-9bc4bb29ee6e
